libobs/media-io/audio-io.c

```diff
@@ -227,18 +230,31 @@ static void *audio_thread(void *param)
 	while (os_event_try(audio->stop_event) == EAGAIN) {
 		uint64_t cur_time;
 
-		os_sleep_ms(audio_wait_time);
+		if (wait_cycles > 0) {
```

Comment:
I don't fully understand the purpose of these changes here:

* At first the thread waits the normal amount of time and then calls `input_and_output` to check if the expected number of samples were provided by the audio source
* If that is the case, the timestamps are updated and the thread continues as before
* If that is **not** the case, the thread waits for _half_ is usual cycle, then increments `wait_cycles` (should be `1` before the next iteration)
* On the next iteration, the thread waits for _half_ the cycle again, then decrements `wait_cycles` again (it should be `0` then).

As such it only seems to wait for half a circle to increment and decrement a single variable that otherwise has no impact on the code executed in the loop. Or am I missing something?